File: mobile/src/debug/java/jp/satorufujiwara/kotlin/AppComponent.java
package jp.satorufujiwara.kotlin;

import dagger.Component;
import jp.satorufujiwara.kotlin.data.DebugDataModule;

@AppScope
@Component(
        modules = {
                DebugAppModule.class,
                DebugDataModule.class
        })
public interface AppComponent extends BaseAppComponent {

    void inject(DaggerApp app);

    static class Initializer {

        private Initializer() {
        }

        public static AppComponent init(DaggerApp app) {
            return DaggerAppComponent.builder()
                    .debugAppModule(new DebugAppModule(app))
                    .debugDataModule(new DebugDataModule())
                    .build();
        }
    }

}

File: mobile/src/debug/kotlin/jp/satorufujiwara/kotlin/AppComponent.kt
package jp.satorufujiwara.kotlin

import dagger.Component
import jp.satorufujiwara.kotlin.data.DebugDataModule

@AppScope
@Component(modules = arrayOf(DebugAppModule::class, DebugDataModule::class))
public interface AppComponent : MainAppComponent {

    fun inject(app: KotlinApp)

    object Initializer {
        fun init(app: KotlinApp): AppComponent =
                DaggerAppComponent.builder()
                        .debugAppModule(DebugAppModule(app))
                        .debugDataModule(DebugDataModule())
                        .build()
    }
}
